### PR TITLE
Fix "find an officer" page

### LINKED ISSUE #196: Fix "find an officer" form back button
https://github.com/lucyparsons/OpenOversight/issues/423

### PR DESCRIPTION
**Pull request by sea-kelp**
## Description of Changes
Fixes #196.

Issues fixed:
* Going back and changing the department didn't reload the unit/rank dropdowns
* "Next" button didn't show up when going back to previous screens

Improvements:
* Added "Previous Step" button
* Disable later steps if department is changed (so user has to re-check their inputs)

## Notes for Deployment
None!

## Tests and linting

 - [x] I have rebased my changes on `main`

 - [x] `just lint` passes

 - [x] `just test` passes

### CODE REVIEW

File: OpenOversight/app/static/js/find_officer.js [hunk 1]
```diff
@@ -1,99 +1,95 @@
...
-            return element.id == $deptSelectionId
+        const deptsWithUii = $('#current-uii').data('departments');
+        let targetDept = deptsWithUii.find(function(element) {
+            return element.id == deptId
         });
-        let targetDeptUii = targetDept.unique_internal_identifier_label
-        if (targetDeptUii) {
-            $('#current-uii').text(targetDeptUii);
+
+        let deptUiidLabel= targetDept.unique_internal_identifier_label
```
Comment (AetherUnbound) [inline OpenOversight/app/static/js/find_officer.js:64]:
```suggestion
        let deptUiidLabel = targetDept.unique_internal_identifier_label
```

**Review by AetherUnbound — COMMENTED** [review]
This is great! I have a couple of notes

- I made an empty department locally and tried to run a search on it. When I got to the Rank/Unit section, there were two options for "Not Sure" under rank:
image

- When I was on Step 2, I hit enter after typing in a name. Instead of taking me to the next page, this actually submitted the form. Is that expected behavior? Should we change that?

**Comment by sea-kelp** [comment]
>     * I made an empty department locally and tried to run a search on it. When I got to the Rank/Unit section, there were two options for "Not Sure" under rank:

I'm seeing this when I check out the main branch too so I'm not sure this is related to my change. Would you mind double-checking on your end?

>     * When I was on Step 2, I hit enter after typing in a name. Instead of taking me to the next page, this actually submitted the form. Is that expected behavior? Should we change that?

Good catch! This should be fixed now

File: OpenOversight/app/static/js/find_officer.js (hunk 1 above)
Comment (sea-kelp) [reply OpenOversight/app/static/js/find_officer.js]: Included in the most recent commit!

**Review by AetherUnbound — APPROVED** [review]
Awesome, thanks for addressing that issue!

**Pull request merged**